transifex/native/tools/migrations/gettext.py

```diff
@@ -0,0 +1,871 @@
+
+            return [''.join([str(x) for x in items])], confidence
+
+        # String with context
+        if gettext_type == PGETTEXT:
+            string, string_quote = new_arguments[KEYWORD_STRING]
+            context, quote_context = new_arguments[KEYWORD_CONTEXT]
+            string = replace_quotes('"{}"', string_quote).format(string)
+            context = replace_quotes('_context="{}"', quote_context).format(
+                context
+            )
+            return [string, context], Confidence.HIGH
+
+        raise ValueError(
+            'Invalid gettext type: {}. Cannot serialize arguments'.format(
+                gettext_type
+            )
+        )
+
+
+def render_keyword(keyword):
+    """Render the given keyword to a proper value.
+
+    Processes Keyword objects and returns its value, using the correct property
+    depending on its type.
+
+    :param Keyword keyword: the keyword object
+    :return: the keyword value (string, number, etc)
+    :rtype: object
+    """
+    if isinstance(keyword, ast.Str):
+        return keyword.s
+    if isinstance(keyword, ast.Num):
+        return keyword.n
+    if isinstance(keyword, ast.Name):
+        return keyword.id
+    return ''
+
+
+def original_quote(node):
+    """Return the quote (' or ") found in the first character of the
+    given node.
+
+    :param ast.Node node: the node to check
+    :return: a 1-char string, ' or "
+    :rtype: str
+    """
+    quote = node.last_token.string[0]
+    return quote if quote in ('"', "'") else None
+
+
+def replace_quotes(string, existing_quote):
+    """Return new string with alternate quotes, if necessary.
+
+    Useful for wrapping strings with the same quotes as those before,
+    if possible, otherwise use alternate quotes.
+
+    :param unicode string: the string that will be wrapped in `outside_quote`
+    :param unicode existing_quote: either ' or ", the quotes used to wrap
+        the `string` when it was parsed
+    :return: a new string
+    :rtype: unicode
+    """
+    if not existing_quote:
+        return string
+    if existing_quote == '"':
+        return string
+    return string.replace('"', "'")
+
+
+def alt_quote(quote, string):
+    """Return the proper quote character to use for wrapping the given string.
+
+    Takes into account any identical quotes that are included in the string
+    and, if found, it returns the alternate quote type, to avoid breaking the
+    code.
+
+    :param unicode quote: either ' or ", the preferred quote to use for wrapping
+        `string`
+    :param unicode string: the string that will be wrapped in quotes
+    :return: a new string
+    :rtype: unicode
+    """
+    alternate = '"' if quote == "'" else "'"
+    pattern = RE_SINGLE_QUOTE if quote == "'" else RE_DOUBLE_QUOTE
+    pattern_alt = RE_SINGLE_QUOTE if quote == '"' else RE_DOUBLE_QUOTE
+    if re.search(pattern, string) and not re.search(pattern_alt, string):
+        return alternate
+    return quote
```

Comment:
That is a really long file. I understood most of the things. Despite it looks fine I think this should be split into a couple of distinct files (e.g. the last three functions may be considered utility ones).

Reply:
It is really long, yes. I'm not happy about this, however I don't see much that we can do at this point.

I'm moving `alt_quote` to a new file (`transifex/common/strings.py`) but the rest are very specific to what we are doing in the migration, despite their "common" look. I don't see them fitting in another module.

Also, I've moved `replace_quotes` as an inner function in `_serialize_arguments`, because it is super specific.

Reply:
@dbendilas I agree with that updates!